parser.go

```diff
@@ -0,0 +1,12 @@
+package editorconfig
+
+import (
+	"gopkg.in/ini.v1"
+)
+
+// Parser interface is responsible for the parsing of the ini file and the
+// globbing patterns.
+type Parser interface {
+	ParseIni(string) (*ini.File, error)
```

Comment:
Should ask for an `io.Reader` instead of an `string`?

Also, I feel that we shouldn't expose `*ini.File` to the public. It would make it impossible to replace this library with another one in the future (while keeping it backward compatible).

Reply:
These comments applies for other places where this was added.

Reply:
The link to the _ini_ library is already there, working around it is doable but requires some amount of extra code. I agree that it would make a lot of sense to abstract it away.

https://godoc.org/github.com/editorconfig/editorconfig-core-go#Definition.InsertToIniFile

Reply:
@andreynering it's way too much work to abstract it and the benefits are quite unclear to me? Do you see any?

Reply:
> Should ask for an io.Reader instead of an string?

The goal is to cache the parsed result of the _filename_, not its content.

Reply:
> @andreynering it's way too much work to abstract it and the benefits are quite unclear to me? Do you see any?

The benefit is having a cleaner API, without references to external libraries.

I know we already have one reference, but I'd love to get rid of it in the next major version (not that this alone is enough to release a new major version, but when it happen...).

Regarding this PR, you choose how to proceed...

Reply:
@andreynering I can only agree with you. I'll take this home for a while and come back later with something nicer. The proposed interface is not good enough.